File: src/main/java/pw/vodes/rimuru/command/commands/CommandUpdate.java
package pw.vodes.rimuru.command.commands;

import org.javacord.api.event.message.MessageCreateEvent;

import pw.vodes.rimuru.Main;
import pw.vodes.rimuru.command.Command;
import pw.vodes.rimuru.command.CommandType;

public class CommandUpdate extends Command {

	public CommandUpdate() {
		super("Update", new String[] {"update"}, CommandType.admin);
	}

	@Override
	public void run(MessageCreateEvent event) {
		if(Main.getUpdater().run(event)) {
			Main.getUpdater().restart();
		}
	}

}

File: src/main/kotlin/pw/vodes/rimurukt/command/commands/CommandUpdate.kt
package pw.vodes.rimurukt.command.commands

import org.javacord.api.event.message.MessageCreateEvent
import pw.vodes.rimurukt.command.Command
import pw.vodes.rimurukt.command.CommandType
import pw.vodes.rimurukt.updater.Updater

class CommandUpdate : Command("Update", arrayOf("update"), CommandType.ADMIN) {
    override fun run(event: MessageCreateEvent) {
        if (!Updater.buildGit(event))
            event.channel.sendMessage("Failed to build. Please check the logs.").also { return }

        if (!Updater.restart())
            event.channel.sendMessage("Failed to restart. Please check the logs.").also { return }
    }
}